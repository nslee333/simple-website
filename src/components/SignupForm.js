export default function SignupForm() {
  return (
    <div className="grid  w-[100%] rounded-md bg-neutral-300 p-2">
      <form
        className="flex-wrap"
        name="newsletter"
        data-netlify="true"
        method="post"
      >
        <input
          className="m-2 block rounded-md p-2"
          type="hidden"
          name="form-name"
          value="newsletter"
        />
        <label>
          <div hidden>name</div>
          <input
            className="m-2 block w-[95%] min-w-[13.25rem] rounded-md p-2"
            type="text"
            name="name"
            placeholder="Full Name"
            required={true}
          />
        </label>
        <label>
          <div hidden>email</div>
          <input
            className="m-2 block w-[95%] min-w-[13.25rem] rounded-md p-2"
            type="email"
            name="email"
            placeholder="Email"
            required={true}
          />
        </label>
        <button
          className="shadow-grey 
                      mb-2 
                      ml-2 
                      mt-4
                      block 
                      h-[2rem] 
                      w-[20%] 
                      min-w-[7rem] 
                      rounded-md 
                      bg-white 
                      text-xl 
                      shadow 
                      hover:origin-bottom 
                      hover:translate-y-0.5
                      hover:shadow-none"
          type="submit"
        >
          Sign up
        </button>
      </form>
    </div>
  );
}
