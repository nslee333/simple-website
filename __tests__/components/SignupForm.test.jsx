import { render } from "@testing-library/react";
import SignupForm from "../../src/components/SignupForm";
import { toBeInTheDocument } from "@testing-library/jest-dom";
import renderer from "react-test-renderer";

test("SignupForm renders without crashing", () => {
  render(<SignupForm />);
});

test("SignupForm renders correctly according to snapshot.", () => {
  const tree = renderer.create(<SignupForm />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Name input is present", () => {
  const { getByLabelText } = render(<SignupForm />);

  const nameInput = getByLabelText("name");

  expect(nameInput).toBeInTheDocument();
});

test("Email input is present", () => {
  const { getByLabelText } = render(<SignupForm />);

  const emailInput = getByLabelText("name");

  expect(emailInput).toBeInTheDocument();
});

test("Check if Sign up button is present", () => {
  const { getByText } = render(<SignupForm />);

  const signUpButton = getByText("Sign up");

  expect(signUpButton).toBeInTheDocument();
});
