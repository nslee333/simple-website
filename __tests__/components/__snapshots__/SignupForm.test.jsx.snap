// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm renders correctly according to snapshot. 1`] = `
<div
  className="grid  w-[100%] rounded-md bg-neutral-300 p-2"
>
  <div
    className="flex-wrap"
    data-netlify="true"
    method="post"
    name="newsletter"
  >
    <input
      className="m-2 block rounded-md p-2"
      name="form-name"
      type="hidden"
      value="newsletter"
    />
    <label>
      <div
        hidden={true}
      >
        name
      </div>
      <input
        className="m-2 block w-[95%] min-w-[13.25rem] rounded-md p-2"
        name="name"
        placeholder="Full Name"
        required={true}
        type="text"
      />
    </label>
    <label>
      <div
        hidden={true}
      >
        email
      </div>
      <input
        className="m-2 block w-[95%] min-w-[13.25rem] rounded-md p-2"
        name="email"
        placeholder="Email"
        required={true}
        type="email"
      />
    </label>
    <button
      className="shadow-grey  mb-2  ml-2  mt-4 block  h-[2rem]  w-[20%]  min-w-[7rem]  rounded-md  bg-white  text-xl  shadow  hover:origin-bottom  hover:translate-y-0.5 hover:shadow-none"
      type="submit"
    >
      Sign up
    </button>
  </div>
</div>
`;
